src/core/engine/processors/java_etl_mongo_order_status_histories_processor.py

```diff
@@ -0,0 +1,97 @@
+from bson import ObjectId
+
+from src.constants.etl_status import Service, EtlStatus
+from src.constants.exception import JwtException
+from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.core.engine.processors.etl.mongo_order_status_history_distribution_etl import (
+    UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT,
+)
+from src.env_config import get_env_config_instance
+from src.grpc.clients.ea_restaurant_java_etl_grpc_client import (
+    EaRestaurantJavaEtlGrpcClient,
+)
+from src.grpc.clients.grpc_client import GrpcClient
+from src.lib.repositories.impl_no_sql.order_status_history_repository_impl import (
+    OrderStatusHistoryRepositoryImpl as MongoOrderStatusHistoryRepository,
+)
+
+UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT = 100
+
+
+class JavaEtlMongoOrderStatusHistoriesProcessor(AbstractProcessor):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+
+        self.mongo_order_status_history_repository = None
+
+        self.env_config = get_env_config_instance()
+        self.grpc_client = GrpcClient(
+            self.env_config.ea_restaurant_java_etl_grpc_host,
+            self.env_config.ea_restaurant_java_etl_grpc_server_port,
+        )
+        self.ea_restaurant_java_etl_grpc_client = EaRestaurantJavaEtlGrpcClient(
+            self.grpc_client
+        )
+        self.access_token = None
+        self.refresh_token = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_order_status_history_repository = ioc.get(
+            MongoOrderStatusHistoryRepository
+        )
+
+    def process(self, delta_time):
+        mongo_order_status_histories = (
+            self.get_unprocessed_java_etl_mongo_order_status_histories()
+        )
+
+        if mongo_order_status_histories:
+            self.send_mongo_order_status_histories_to_java_etl(
+                mongo_order_status_histories
+            )
+
+    def get_unprocessed_java_etl_mongo_order_status_histories(self):
+        mongo_order_status_histories = self.mongo_order_status_history_repository.get_order_status_histories_by_service(
+            Service.JAVA_ETL,
+            EtlStatus.UNPROCESSED,
+            limit=UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT,
+        )
+        return mongo_order_status_histories
+
+    def send_mongo_order_status_histories_to_java_etl(
+        self, mongo_order_status_histories
+    ):
+        def run_grpc_call():
```

Comment:
why this is there?

This logic should inside the `ea_restaurant_java_etl_grpc_client`

Reply:
Done.